src/components/ProjectList/CardsContainer.jsx

```diff
@@ -15,125 +15,146 @@ export default class CardsContainer extends React.Component {
-
-          searchedList.push(project)
+      const searchedList = [];
+      each(
+        updatedList.length > 0 ? updatedList : projectList,
+        (project) => {
+          if (
+            project.name.toLowerCase().includes(this.inputValue) ||
+            project.description.toLowerCase().includes(this.inputValue) ||
+            project.tags.toString().toLowerCase().includes(this.inputValue)
+          ) {
+            searchedList.push(project);
+          }
         }
-      });
+      );
 
       updatedList = searchedList;
     }
 
-    this.setState({ filterList: updatedList });
+    // Update state with results or error message
+    if (updatedList.length > 0) {
+      this.setState({ filterList: updatedList, error: "" });
+    } else {
+      this.setState({
+        filterList: [],
+        error: `No results found for your search: "${this.inputValue}". Please try again.`,
+      });
+    }
   }
 
   // Search input handler
   handleChange(event) {
-
-    this.inputValue = event.currentTarget.value;
-
-    this.inputValue = this.inputValue.trim();
-    this.inputValue = this.inputValue.toLowerCase();
-
-    this.handleFilterListUpdate(this.value)
+    this.inputValue = event.currentTarget.value.trim().toLowerCase();
+    this.handleFilterListUpdate(this.value);
   }
 
-  sortArrayRandom(array){
-    if(Array.isArray(array)){
-      return array.sort(()=>0.5-Math.random())
+  sortArrayRandom(array) {
+    if (Array.isArray(array)) {
+      return array.sort(() => 0.5 - Math.random());
     }
-    return array
+    return array;
   }
 
   render() {
-
     return (
       <div>
-        <div id='container'>
-          <div className='inputContainer'>
-            <input id='search' type='text' name='search' placeholder='Search...' onChange={this.handleChange} aria-label='Search'/>
+        <div id="container">
+          <div className="inputContainer">
+            <input
+              id="search"
+              type="text"
+              name="search"
+              placeholder="Search..."
+              onChange={this.handleChange}
+              aria-label="Search"
+            />
           </div>
-          <div id="tag-selector-container" className='inputContainer'>
+          <div id="tag-selector-container" className="inputContainer">
             <Select
-              name='tag-selector'
+              name="tag-selector"
               value={this.state.value}
               onChange={this.handleSelectChange}
               options={this.filterOptions}
               multi={true}
-              placeholder={<div className='filter-placeholder-text'>Filter</div>}
-              aria-labelledby='tag-selector-container'
+              placeholder={<div className="filter-placeholder-text">Filter</div>}
+              aria-labelledby="tag-selector-container"
             />
           </div>
         </div>
-        <section id='project-list' className='containerLayout'>
+
+        {this.state.error && (
```

Comment:
What about showing error messages only if there is nothing in the `filterList`? Then you don't need another state like `error`. Maybe that could work 🤷